plugins/serializers/splunkmetric/splunkmetric.go

```diff
@@ -0,0 +1,125 @@
+package splunkmetric
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+)
+
+type serializer struct {
+	HecRouting bool
+}
+
+func NewSerializer(hec_routing bool) (*serializer, error) {
+	s := &serializer{
+		HecRouting: hec_routing,
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+
+	m, err := s.createObject(metric)
+	if err != nil {
+		log.Printf("D! [serializer.splunkmetric] Dropping invalid metric: %v [%v]", metric, m)
+		return nil, nil
+	}
+
+	return m, nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized []byte
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err != nil {
+			log.Printf("D! [serializer.splunkmetric] Dropping invalid metric: %v [%v]", metric, m)
+		} else {
+			serialized = append(serialized, m...)
+		}
+	}
+
+	return serialized, nil
+}
+
+func (s *serializer) createObject(metric telegraf.Metric) (metricJson []byte, err error) {
+
+	/*  Splunk supports one metric json object, and does _not_ support an array of JSON objects.
+	     ** Splunk has the following required names for the metric store:
+		 ** metric_name: The name of the metric
+		 ** _value:      The value for the metric
+		 ** time:       The timestamp for the metric
+		 ** All other index fields become deminsions.
+	*/
+	type HECTimeSeries struct {
+		Time   float64                `json:"time"`
+		Event  string                 `json:"event"`
+		Host   string                 `json:"host,omitempty"`
+		Index  string                 `json:"index,omitempty"`
+		Source string                 `json:"source,omitempty"`
+		Fields map[string]interface{} `json:"fields"`
+	}
+
+	dataGroup := HECTimeSeries{}
+
+	for k, v := range metric.Fields() {
+
+		if !verifyValue(v) {
+			log.Printf("D! Can not parse value: %v for key: %v", v, k)
```

Comment:
I think here we should just continue on the next field.  Otherwise we will never be able to serialize a metric with a string field.

Reply:
Got it. Will include in the other changes being made.